src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -32,6 +33,19 @@
 
   private DatastoreService datastore;
 
+  /** 
+   *  Checks if a user was already stored, done when a user logs in.
+   * 
+   *  @return true if a user in Datastore, false otherwise.
+   */
+  private boolean userFound(String user) {
```

Comment:
Not a comment on this change specifically, but you will soon run into a terminology problem between "user" (the string used to refer to a user entity), "User" (the entity or data model object you are thinking about creating) and "user" (a variable you'd use in the future to refer to a user *entity* instance).

In particular, the issue is caused entirely by "String".  A "String" could be literally anything, so you are calling this particular String "user" just because it represents a handle to the user's data.  However, userId would be more clear in the future (don't change it yet, I'm just giving you a head's up).  This problem is so bad that a lot of times one would just create a class called UserId which basically is just a String, so that you can make your interfaces more typesafe but simpler, as in lookUp(UserId) -> Entity versus lookUpUserById(String) -> UserEntity

Reply:
Ah I see how it could get bad. Thanks for the early heads up